src/components/DocumentsView/DocumentView.tsx

```diff
@@ -17,21 +17,34 @@ export const DocumentView: FunctionComponent<IDocumentViewProps> = (props) => {
     const [deleteTarget, setDeleteTarget] = useState<any>();
 
     const extension = props.document.LinkUrl.substr(props.document.LinkUrl.lastIndexOf('.') + 1);
+    const wordExtensions = ["doc", "dot", "wbk", "docx", "docm", "dotx", "dotm", "docb"];
+    const excelExtensions = ["xls", "xlt", "xlm", "xlsx", "xlsm", "xltx", "xltm", "xlsb", "xla", "xlam", "xll", "xlw"];
+    const ppExtensions = ["ppt", "pot", "pps", "pptx", "pptm", "potx", "potm", "ppam", "ppsx", "ppsm", "sldx", "sldm"];
 
     const deleteIconOnclick = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
         setDeleteTarget(e.target);
         setShowDeletePopover(true);
     }
 
+    const getDocLink = (): string => {
+        if (wordExtensions.includes(extension)) {
+            return `${window.location.origin}/:w:/r${props.document.LinkUrl.substring(0, props.document.LinkUrl.indexOf("/Processes"))}/_layouts/15/Doc.aspx?sourcedoc={${props.document.UniqueId}}&file=${props.document.Name}&action=default&mobileredirect=true`
+        } else if (excelExtensions.includes(extension)) {
+            return `${window.location.origin}/:x:/r${props.document.LinkUrl.substring(0, props.document.LinkUrl.indexOf("/Processes"))}/_layouts/15/Doc.aspx?sourcedoc={${props.document.UniqueId}}&file=${props.document.Name}&action=default&mobileredirect=true`
+        } else if (ppExtensions.includes(extension)) {
+            return `${window.location.origin}/:p:/r${props.document.LinkUrl.substring(0, props.document.LinkUrl.indexOf("/Processes"))}/_layouts/15/Doc.aspx?sourcedoc={${props.document.UniqueId}}&file=${props.document.Name}&action=default&mobileredirect=true`
+        } else {
+            return props.document.LinkUrl;
+        }
+    }
```

Comment:
Should these open in a new tab?

Reply:
I thought about doing it like that but I was just going by the current functionality that Kelly was describing, she said they would click on the doc, edit it, save it, and then press the back button and it would be updated.